contracts/interchain-token-service/src/contract/migrations/mod.rs

```diff
@@ -1,16 +1,442 @@
-use axelar_wasm_std::migrate_from_version;
+use std::collections::HashMap;
+
+use axelar_wasm_std::address::validate_cosmwasm_address;
+use axelar_wasm_std::{migrate_from_version, IntoContractError};
+use cosmwasm_schema::cw_serde;
 #[cfg(not(feature = "library"))]
 use cosmwasm_std::entry_point;
-use cosmwasm_std::{DepsMut, Empty, Env, Response};
+use cosmwasm_std::{DepsMut, Env, Response, Storage};
+use cw_storage_plus::Map;
+use error_stack::ResultExt;
+use router_api::{Address, ChainNameRaw};
+
+use crate::shared::NumBits;
+use crate::state::{save_chain_config, ChainConfig as NewChainConfig};
+
+// Old state structure (before migration)
+#[cw_serde]
+pub struct OldTruncationConfig {
+    pub max_uint_bits: NumBits,
+    pub max_decimals_when_truncating: u8,
+}
 
-pub type MigrateMsg = Empty;
+#[cw_serde]
+pub struct OldChainConfig {
+    pub truncation: OldTruncationConfig,
+    pub its_address: Address,
+    pub frozen: bool,
+    // Note: no translation_contract field in old state
+}
+
+// Storage map for old chain configs
+const OLD_CHAIN_CONFIGS: Map<&ChainNameRaw, OldChainConfig> = Map::new("chain_configs");
```

Comment:
wrap all of the old types and methods inside a `mod legacy { ... }` so you can access it via `legacy::ChainConfig` instead of having to prefix it with `Old`

example
https://github.com/axelarnetwork/axelar-amplifier-stellar/blob/c86b31621b4b1022f33e6bb5e80358ca6658e8d7/contracts/stellar-interchain-token-service/src/migrate.rs#L19

Reply:
We could even create a `legacy_state.rs` file with the types and methods. Easier to see the diff and safer to keep it separate